typescript/sdk/src/deploy/middleware/LiquidityLayerApp.ts

```diff
@@ -110,24 +153,76 @@ export class LiquidityLayerApp<
         txHash,
         message,
         nonce,
-        domain: 0,
+        domain,
         nonceHash: ethers.utils.solidityKeccak256(
           ['uint32', 'uint256'],
-          [0, nonce],
+          [domain, nonce],
         ),
       },
     ];
   }
 
+  async attemptPortalTransferCompletion(
+    message: PortalBridgeMessage<Chain>,
+  ): Promise<void> {
+    const destinationPortalAdapter = this.getContracts(message.destination)
+      .portalAdapter!;
+
+    const transferId = await destinationPortalAdapter.transferId(
+      ChainNameToDomainId[message.origin],
+      message.nonce,
+    );
+
+    const transferTokenAddress =
+      await destinationPortalAdapter.portalTransfersProcessed(transferId);
+
+    if (!utils.eqAddress(transferTokenAddress, ethers.constants.AddressZero)) {
+      console.log(
+        `Transfer with nonce ${message.nonce} from ${message.origin} to ${message.destination} already processed`,
+      );
+      return;
+    }
+
+    const wormholeOriginDomain = this.config[
+      message.destination
+    ].portal!.wormholeDomainMapping.find(
+      (_) => _.hyperlaneDomain === ChainNameToDomainId[message.origin],
```

Comment:
this is hard to read
make these actual 2 way mappings?

Reply:
I agree it doesn't feel super intuitive, but there are no actual two way mappings in JS no? It seemed like making this into an object, and then dealing with stringified keys was not super great either